### PR TITLE
Extend functionality of debug command and memory map attributes check.

### PR DESCRIPTION
**Pull request by gerard5**
## Description

Extends the functionality of the memory content analysis tool. Performs `memory range checking` against defined maps and whether the `read attribute is set`. If the memory range given in the arguments for the `dump` command is incorrect, or no read access is available, or the maps are not defined at all, the command will display masked octet value `XX` for the given range.
Additionally the selected range is printed starting form alignment down to 16 bytes + offset.

rect851-3-6-7

When used with flag `-F` memory map validation is disabled (than works like the earlier implementation).

### Additional feature

An additional option is to read disks `ia32`  hard disk or floppy disk, on `imxrt-flexspi` via IP Bus it reads NOR memory, utilizing `phfs` subsystem.

Example on ia32: `dump -r hd0 0 100` prints 100 bytes from the beginning of hard drive.

2022-10-24-174126_909x253_scrot

Example on imxrt117x: `dump -F -r flash0 0x11000 0x20` reads 32 bytes of ELF header located somewhere on flash0 reading through IP Bus, not an AHB.

2022-10-24-174339_660x96_scrot

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by agkaminski** [comment]
These invalid accessed caused CPU exception, I guess? Otherwise it might sometimes be useful to peek at memory we're not supposed to peek at.

**Comment by gerard5** [comment]
> These invalid accesses caused CPU exception, I guess? Otherwise it might sometimes be useful to peek at memory we're not supposed to peek at.

That's right, but of course I'll add `a switch to bypass` this, but it will `validate` access with defined `maps by default` like in this PR.

**Review by HBuczynski — DISMISSED** [review]
Nice feature ;)
LGTM.

File: cmds/dump.c [hunk 1]
```diff
@@ -17,61 +17,214 @@
...
+}
+
+
+static void print_hline(void)
+{
+	unsigned int col;
+	for (col = 0; col < 79; ++col) {
+		lib_consolePutc('-');
+	}
+	lib_printf("\n");
```
Comment (agkaminski) [inline cmds/dump.c:111]: Just a suggestion, somewhat faster approach, we don't have to worry about stack in plo that much
```suggestion
	char buff[80];
	hal_memset(buff, '-', sizeof(buff) - 1);
	buff[sizeof(buff) - 1] = '\n';
	lib_printf(buff);
```

**Review by agkaminski — DISMISSED** [review]
One small suggestion, nevertheless LGTM

File: cmds/dump.c (hunk 1 above)
Comment (gerard5) [reply cmds/dump.c:111]: `printf()` [underneath calls](https://github.com/phoenix-rtos/plo/blob/master/lib/printf.c#L25-L29) `lib_consolePutc()` so it's not the fastest and is stack hungry, but well ... I missed that last `lib_printf("\n")` can be replaced with `lib_consolePutc('\n')` :-) Thanks in advance for pointing it out. I'll change it like so, and other calls to `printf()` that does not use formatting I'll change to `lib_consolePuts()`, this is what compiler do for `printf`s if it detects no formating it replaces with puts/putc respectively.

**Review by HBuczynski — APPROVED** [review]
Tested on zedboard. Works as expected.

**Pull request merged**